docs/design/interoperability/README.md

```diff
@@ -29,4 +49,165 @@ more detail.
+interoperability layer is designed to be zero-cost, avoiding unnecessary
+allocations or copies when calling between the two languages.
+
+## C++ interoperability model: introduction and principles
+
+### The "successor language" mandate
+
+The design of Carbon's C++ interoperability is governed by its foundational
+goal: [to be a successor language](/README.md), not merely a language with a
+foreign function interface (FFI). This mandate dictates a design that moves
+beyond the C-style FFI adopted by most modern languages and instead provides
+"seamless, bidirectional interoperability". The objective is to support deep
+integration with existing C++ code, encompassing its most complex features,
+"from inheritance to templates".
+
+This goal has profound implications for the Carbon compiler and language
+semantics. It requires that C++ is not treated as a "foreign" entity. Instead,
+Carbon's semantic model must be _co-designed_ to understand, map, and interact
+with C++'s semantic constructs—including templates, class hierarchies, and
+complex overload resolution—with high fidelity. The interoperability layer must,
+therefore, operate at the semantic analysis level, not just at the linking (ABI)
+level. This document specifies the design of this semantic contract.
+
+### The C++ interop type
+
+A core mechanism in this design is the C++ interop type. This concept defines
+the "trigger" that activates C++-specific semantic rules within the Carbon
+compiler. Any operation involving a type that is designated as a C++ interop
+type could invoke the specialized interoperability logic, such as C++ overload
+resolution or operator overload resolution that involves both Carbon and C++
+operator overloads..
+
+A type is considered a C++ interop type if its definition involves an imported
+C++ type in any of the following ways:
+
+1.  A C++ imported type (for example, `Cpp.Widget`).
+2.  A pointer to a C++ interop type (for example, `Cpp.Widget*`).
+3.  A Carbon generic type parameterized with a C++ interop type (for example,
+    `MyCarbonVector(Cpp.Widget)`).
+4.  A Carbon struct or class containing a C++ interop type as a member (for
+    example, `MyCarbonStruct { x: Cpp.Widget }`).
+
+This "pervasive" model of C++-awareness is a fundamental design choice. The C++
+semantics are not confined to a specific `unsafe` or `extern "C++"` block; they
+affect any Carbon type that composes them. For example, when the Carbon compiler
+instantiates a _Carbon_ generic type like `MyCarbonVector(Cpp.Widget)`, its type
+system must be aware that the `Cpp.Widget` parameter carries C++-specific rules.
+This mandates that Carbon's own generic system, struct layout logic, overload
+resolution and operator lookup must query the type system for the presence of a
+C++ interop type. If present, Carbon must consider C++ rules when operating over
+C++ interop types. This design prioritizes the goal of a "seamless" and
+"intuitive" user experience.
```

Comment:
As above, I'd drop the double-quotes here.

Reply:
Done.